### PR TITLE
Order editing

### PR DESCRIPTION
**Pull request by sfiorucci**
## What I did

Update API reference according to the new order editing feature.

### CODE REVIEW

File: orders/README.md [hunk 1]
```diff
@@ -12,11 +12,12 @@ Orders get created in `draft` status and become `pending` when they have a custo
...
+`pending` orders can be recovered when abandoned. The status of an order is closely linked to the related payment and fulfillment statuses. When an order is `placed`, it can either get `approved` or `cancelled`. An approved order becomes `fulfilled` when paid and shipped. Cancelling an order automatically [voids](../voids/) its payment source's [authorization](../authorizations/). Captured payments can be [refunded](../refunds/), either fully or partially.
 
 | Action                    | Order status | Payment status       | Fulfillment status                                     |
 | ------------------------- | ------------ | -------------------- | ------------------------------------------------------ |
 | **Order placement**       | `placed`     | `authorized`         | `unfulfilled`                                          |
+| **Order editing**         | `editing`    | `authorized`         | `unfulfilled`                                          |
```
Comment (sfiorucci) [inline orders/README.md:20]: Wondering if we need to add, for **Order editing** and **Order approval**, "`authorized` or `paid`" in the _Payment status_ column. On the other hand, I wouldn't add too much noise, this is a reference of the most common use case and we cannot show all the possible conditions/edge cases using a simple table. Open to suggestions.

File: orders/README.md [hunk 2]
```diff
@@ -34,21 +35,43 @@ Read more about the anatomy of an order [here](https://commercelayer.io/docs/dat
...
+
+{% hint style="warning" %}
+For security reasons, only [integration](http://localhost:5000/s/-LgByaSP8eKjad-MIuHE/applications#integration) applications are allowed to change the `_start_editing` and `_stop_editing` attributes of an order. When an order is in the `editing` status it can be updated by the customer that placed it.
+{% endhint %}
+
+When an order is in editing you can make almost any change you need (e.g. adding or removing line items, coupon or gift cards, adjustment, etc.) except replace the payment source and payment method, on the condition that the updates don't lead to a total order amount that exceeds the previously authorized (or [already captured](README.md#capture-options), if that's the case) amount, otherwise the API will return an error on the `_stop_editing`. If, after the order editing, the total amount is less than the authorized amount, the new amount will be captured. If the authorized amount was already captured, a partial [refund](../refunds) will be performed.
+
```
Comment (mkcosta) [inline orders/README.md:47]: Do we want to add a note about (allowed) multiple editing or it's implicit?

File: orders/README.md [hunk 3]
```diff
@@ -34,21 +35,43 @@ Read more about the anatomy of an order [here](https://commercelayer.io/docs/dat
...
+#### Shipments and shipping methods
+
+Shipments are rebuilt so, after any editing operation and before exiting the editing status, you need to check again the [available shipping methods](http://localhost:5000/s/-Lk-ezuDClaMavTqnRi0/placing-orders/checkout/selecting-a-shipping-method#1.-get-the-available-shipping-methods) (which may be different) and choose a new one to be [re-associated](http://localhost:5000/s/-Lk-ezuDClaMavTqnRi0/placing-orders/checkout/selecting-a-shipping-method#3.-select-a-shipping-method) with the edited order, otherwise the API will return an error on the `_stop_editing`.
+
+{% hint style="info" %}
+Orders can be edited multiple times before approval. Please note that once an editing operation is started, it can't be reverted. It can always be aborted by cancelling the order.
```
Comment (sfiorucci) [inline orders/README.md:57]: @mkcosta the "allowed multiple editing" note is already here. 👍

File: orders/README.md (hunk 2 above)
Comment (sfiorucci) [reply orders/README.md:47]: There is already a note about that down below the page.

**Pull request merged**